methodologies/kinesthetic-programming/README.md

```diff
@@ -0,0 +1,81 @@
+# Kinesthetic Programming
+
+by @bandrzejewski
+
+Numbers of developers that are reading documentation in order to understand used code are doomed to failure. The reason behind 
+this is that they need to understand both code and the documentation. Let's be honest, there are only a few developers that
+can write good code and good, explanatory documentation. It is even common, that different persons are writing those parts.
+It is very easy to introduce misleading information causing confusion and misunderstanding. In order to solve this problem,
+Kinesthetic Programming could be used. It reduces the amount of point of failures by simply skipping documentation at all.
+Instead of that, it develops skills that seem to be vanishing among developers, like reading others code, understanding
+it, or ability to debug the code.
+
+Kinesthetic Programming is based on a few fundamentals:
+* code, that you are using for implementing a feature should have meaningful naming. Methods, that you are using, should be descriptive but compact
+* quick feedback loop - it is the best if you have a set of unit tests that can give you a response in milliseconds, but any test suite that executes faster than you unlock your smartphone will go
+* access to source codes of classes and libraries that you are using
+* IDE with nice code assistant
+
```

Comment:
You maybe wonder, what are the 'secret' techniques of kinesthetic programming. Lets take a look at few examples and describe some behaviours.
Did you ever noticed, when running your tests, some weird exception in the logs? Or did you ever tried to introduce some new fancy technology, with cool name - but somehow it did not jump started right away? I know you did. Everyone of us know this feeling. That are the situations where Kinesthetic comes into play. One of the techniques is CAR - Change and Run. It is all about 'feeling' the code. You just try to change some pieces of code/ configuration - that you think will fix it - and give it a try. If not succeeded, introduce another change and repeat the process until success. By following CAR technique, we nee to focus on the practice and we try to skip documentation reading part (which is often incomplete/ or just really boring).
Another sole part of the techniques is the way how we look the answers in the internet. Everyone from time to time (Ok, mostly) looks for solution by simply googling it. Thats perfectly fine, but how you use the articles/documentation is not so obvious. You might want to read them whole - usually few pages of reading - but let's be hones, who writes that, it's usually rubbish. Whole clue of the reading is usually somewhere in the bottom - yes - The Snippets! What kinesthetic encourages you to do is to go straight away to the meritum, just copy The Snippets and give it a try. That will probably save you a lot of time. It usually works, but sometimes the snippets does not fit right into our projects - you might ask what then? It is perfectly fine to open another article and give it a try - follow kinesthetic approach! If you feel confident, you can even try to combine snippets from different articles - but that will require some experience. You you are adept, better not to overengineer it and stick to one.